src/Dialogs/Cloud/WeGlideUploadDialog.cpp

```diff
@@ -0,0 +1,191 @@
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "WeGlideUploadDialog.hpp"
+#include "Interface.hpp"
+#include "Cloud/weglide/WeGlideSettings.hpp"
+#include "Cloud/weglide/UploadFlight.hpp"
+#include "Operation/PluggableOperationEnvironment.hpp"
+#include "net/http/Init.hpp"
+#include "json/ParserOutputStream.hxx"
+#include "Dialogs/CoDialog.hpp"
+#include "Dialogs/Error.hpp"
+#include "UIGlobals.hpp"
+#include "Language/Language.hpp"
+#include "co/InvokeTask.hxx"
+#include "Dialogs/Message.hpp"
+#include "util/Exception.hxx"
+#include "util/ConvertString.hpp"
+#include "io/StringConverter.hpp"
+#include "LogFile.hpp"
+#include "json/Lookup.hxx"
+#include "system/Path.hpp"
+#include "LocalPath.hpp"
+#include "system/FileUtil.hpp"
+
+#include <inttypes.h>
+
+namespace WeGlide {
+
+  typedef struct _Glider {
+    uint64_t id;
+    StaticString<0x40> name;
+    StaticString<4> kind;  // 'MG' - motor glider,...
+    StaticString<10> sc_class;
+    uint64_t seats;
+  } Glider;
+
+  typedef struct _UploadResponse {
+    uint64_t flight_id;
+    Pilot pilot;
+    Glider glider;
+    // StaticString<0x40> pilot_name;
+    StaticString<0x40> scoring_date;
+    StaticString<0x40> registration;
+    StaticString<0x40> competition_id;
+  } UploadResponse;
+
+  static Co::InvokeTask UpdateTask(Path igc_path, const char* uri,
+    const WeGlide::Pilot& pilot,
+    uint_least32_t glider_id,
+    UploadResponse& response,
+    ProgressListener& progress) noexcept {
+    boost::json::value json =
+      co_await WeGlide::UploadFlight(*Net::curl, uri, pilot, glider_id,
+        igc_path, progress);
+    auto json_value = json.get_array().at(0);
+    response.scoring_date = UTF8ToWideConverter(
+      Json::Lookup(json_value, "scoring_date")->get_string().c_str());
+    response.flight_id = Json::Lookup(json_value, "id")->get_int64();
+    response.registration = UTF8ToWideConverter(
+      Json::Lookup(json_value, "registration")->get_string().c_str());
+    response.competition_id = UTF8ToWideConverter(
+      Json::Lookup(json_value, "competition_id")->get_string().c_str());
+    auto user = Json::Lookup(json_value, "user")->as_object();
+
+    response.pilot.id =
+      Json::Lookup(user, "id")->get_int64();
```

Comment:
This will crash if the "id" field is not present.

Reply:
Solved with a wrapper function...